utils/cLog.cpp

```diff
@@ -1,262 +1,204 @@
-
-cLogger::cLogger() {
-	startTime = clock();
-}
-
-cLogger::~cLogger() {
-}
-
-cLogger *cLogger::getInstance() {
-	if (instance == NULL) {
-		instance = new cLogger();
-		logbook("New instance of Logger created");
-	}
-
-	return instance;
-}
-
-std::string cLogger::getLogLevelString(eLogLevel level) {
-	// LOG_TRACE, LOG_WARN, LOG_ERROR, LOG_FATAL
-	switch (level) {
-		case LOG_FATAL:
-			return std::string("FATAL");
-		case LOG_WARN:
-			return std::string("WARN");
-		case LOG_ERROR:
-			return std::string("ERROR");
-		case LOG_TRACE:
-			return std::string("TRACE");
-		case LOG_INFO:
-			return std::string("INFO");
-	}
-
-	return std::string("UNIDENTIFIED");
-}
-
-std::string cLogger::getLogComponentString(eLogComponent component) {
-	switch(component) {
-		case COMP_UNITS:
-			return std::string("UNITS");
-		case COMP_STRUCTURES:
-			return std::string("STRUCTURES");
-		case COMP_GAMEINI:
-			return std::string("GAMEINI");
-		case COMP_SCENARIOINI:
-			return std::string("SCENARIOINI");
-		case COMP_PARTICLE:
-			return std::string("PARTICLE");
-		case COMP_BULLET:
-			return std::string("BULLET");
-		case COMP_AI:
-			return std::string("AI");
-    case COMP_UPGRADE_LIST:
-      return std::string("UPGRADE_LIST");
-	  case COMP_BUILDING_LIST_UPDATER:
-      return std::string("BUILDING_LIST_UPDATER");
-		case COMP_MAP:
-			return std::string("MAP");
-		case COMP_SIDEBAR:
-			return std::string("SIDEBAR");
-		case COMP_NONE:
-			return std::string("NONE");
-		case COMP_SETUP:
-			return std::string("SETUP");
-		case COMP_INIT:
-			return std::string("INIT");
-		case COMP_ALLEGRO:
-			return std::string("ALLEGRO");
-		case COMP_VERSION:
-			return std::string("VERSION");
-		case COMP_SKIRMISHSETUP:
-			return std::string("SKIRMISHSETUP");
-		case COMP_ALFONT:
-			return std::string("ALFONT");
-		case COMP_SOUND:
-			return std::string("SOUND");
-		case COMP_REGIONINI:
-			return std::string("REGIONINI");
-	}
-
-	return std::string("UNIDENTIFIED");
-}
-
-std::string cLogger::getLogOutcomeString(eLogOutcome outcome) {
-	switch (outcome) {
-		case OUTC_SUCCESS:
-			return std::string("SUCCESS");
-		case OUTC_FAILED:
-			return std::string("FAILED");
-		case OUTC_NONE:
-			return std::string("NONE");
-	  case OUTC_UNKNOWN:
-      break;
-	  case OUTC_IGNOREME:
-      break;
-	}
- 	return std::string("UNIDENTIFIED");
-}
-
-std::string cLogger::getLogHouseString(int houseId) {
-	switch (houseId) {
-		case ATREIDES:
-			return std::string("ATREIDES");
-		case HARKONNEN:
-			return std::string("HARKONNEN");
-		case ORDOS:
-			return std::string("ORDOS");
-		case FREMEN:
-			return std::string("FREMEN");
-		case SARDAUKAR:
-			return std::string("SARDAUKAR");
-		case MERCENARY:
-			return std::string("MERCENARY");
-		default:
-			return std::string("UNKNOWN HOUSE");
-	}
-	return std::string("UNIDENTIFIED");
-}
-
-
-// courtesy from : http://www.codeguru.com/forum/showthread.php?t=477894
-std::string cLogger::getCurrentFormattedTime() {
```

Comment:
If we want this functionality in the future, I can recommend the standard chrono library.